templates/repo/diff/compare.tmpl

```diff
@@ -187,8 +187,21 @@
 		{{end}}
 	{{else if and .PageIsComparePull (gt .CommitCount 0)}}
 		{{if .HasPullRequest}}
-			<div class="ui segment">
-				{{.i18n.Tr "repo.pulls.has_pull_request" (Escape $.RepoLink) (Escape $.RepoRelPath) .PullRequest.Index | Safe}}
+			<div class="ui segment grid title">
+				<div class="twelve wide column issue-title">
+					<h1>
+						<span id="issue-title">{{RenderIssueTitle .PullRequest.Issue.Title $.RepoLink $.Repository.ComposeMetas}}</span>
+						<span class="index">#{{.PullRequest.Issue.Index}}</span>
+					</h1>
+				</div>
+				<div class="four wide right aligned column">
+				{{- if .PullRequest.HasMerged -}}
+				<a href="{{Escape $.RepoLink}}/pulls/{{.PullRequest.Issue.Index}}" class="ui button purple show-form">{{svg "octicon-git-merge" 16}} {{.i18n.Tr "repo.pulls.view"}}</a>
+				{{else if .Issue.IsClosed}}
+				<a href="{{Escape $.RepoLink}}/pulls/{{.PullRequest.Issue.Index}}" class="ui button red show-form">{{svg "octicon-issue-closed" 16}} {{.i18n.Tr "repo.pulls.view"}}</a>
+				{{else}}
+				<a href="{{Escape $.RepoLink}}/pulls/{{.PullRequest.Issue.Index}}" class="ui button green show-form">{{svg "octicon-git-pull-request" 16}} {{.i18n.Tr "repo.pulls.view"}}</a>
+				{{end}}</div>
```

Comment:
Do we need a template to render the PR icon?
I'd prefer so.

Reply:
If no other place can reuse it, I don't think we should move it to a new template. For different state of pull request, the color of the button and the icon are not different.